for_workspace/make_build.bzl

```diff
@@ -1,28 +1,41 @@
 """ Rule for building GNU Make from sources. """
 
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load(
+    "@rules_foreign_cc//tools/build_defs:run_shell_file_utils.bzl",
+    "fictive_file_in_genroot",
+)
 load("@rules_foreign_cc//tools/build_defs:shell_script_helper.bzl", "convert_shell_script")
 load("//tools/build_defs:detect_root.bzl", "detect_root")
 
 def _make_tool(ctx):
     root = detect_root(ctx.attr.make_srcs)
 
+    cc_toolchain = find_cpp_toolchain(ctx)
+
+    # we need this fictive file in the root to get the path of the root in the script
+    empty = fictive_file_in_genroot(ctx.actions, ctx.label.name)
```

Comment:
Can you elaborate a bit more on what's happening here?

Reply:
This is a copy of what is done in `framework.bzl` basically its creating an empty output file so that the script can detect the folder to use for INSTALLDIR.